easybuild/tools/filetools.py

```diff
@@ -495,7 +495,13 @@ def pypi_source_urls(pkg_name):
         _log.debug("Failed to download %s to determine available PyPI URLs for %s", simple_url, pkg_name)
         res = []
     else:
-        parsed_html = ElementTree.parse(urls_html)
+        urls_txt = read_file(urls_html)
+
+        # strip out data-yanked attributes before parsing HTML
+        # see https://github.com/easybuilders/easybuild-framework/issues/3301
+        urls_txt = re.sub(r'\s*data-yanked', '', urls_txt)
```

Comment:
According to PyPI directives, yanked releases have to be ignored by installers. So we should completely remove those entries from the list
```suggestion
        urls_txt = re.sub('^.*data-yanked.*$', '', urls_txt, flags=re.M)
```

Reply:
OK, fixed in 5409a27 (also updated comment)